src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -4687,6 +4972,19 @@ protected override T GetMember<T>(object obj, string memberName)
             return ((PSMemberSet)obj).Members[memberName] as T;
         }
 
+        private protected override T GetMember<T>(object obj, MemberNamePredicate predicate)
+        {
+            foreach (var mem in ((PSMemberSet)obj).Members)
+            {
+                if (predicate(mem.Name))
```

Comment:
```suggestion
                return predicate(mem.Name)) ? mem as T : null;
```
(Remember to replace the body of the if block if you take the suggestion.)